### PR TITLE
Create an interstitial to log in.

### PR DESCRIPTION
**Pull request by ndmckinley**
### Description
When a user clicks through the summary page, submit the application, then consider asking the user if they should sign in.

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
Implements #853.
Screen Shot 2021-05-14 at 6 35 49 PM

### CODE REVIEW

File: universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramReviewController.java [hunk 1]
```diff
@@ -157,20 +157,16 @@ public ApplicantProgramReviewController(
...
     return submitApp
-        .thenComposeAsync(
-            application -> applicantService.getReadOnlyApplicantProgramService(application),
-            httpExecutionContext.current())
         .thenApplyAsync(
-            roApplicantProgramService -> {
+            application -> {
               // This must already be done since we are behind it in the promise chain
-              Long applicationId = submitApp.toCompletableFuture().join().id;
+              Long applicationId = application.id;
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramReviewController.java:163]: Nit (non-blocking): would you remove the comment above?

File: universal-application-tool-0.0.1/app/controllers/applicant/RedirectController.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+              // Don't show this page to TIs, or anyone with an email address already.
+              if (!Strings.isNullOrEmpty(account.getEmailAddress())) {
+                return redirect(redirectTo);
+              } else if (account.getMemberOfGroup().isPresent()) {
+                return redirect(redirectTo);
+              }
+
+              // To make this happen we really need to be pretty sure we are able to redirect
+              // correctly
+              // after a login!
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/app/controllers/applicant/RedirectController.java:84]: Nit (blocking): mind fixing the line break?

File: universal-application-tool-0.0.1/app/controllers/applicant/RedirectController.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                return notFound();
+              }
+              return redirect(
+                  controllers.applicant.routes.ApplicantProgramsController.edit(
+                      applicant.join().id, program.join().id));
+            });
+  }
+
+  @Secure
+  public CompletableFuture<Result> considerSignIn(Http.Request request, String redirectTo) {
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/app/controllers/applicant/RedirectController.java:63]: Suggestion (non-blocking): should this be considerRegister? or considerSignUp?

File: universal-application-tool-0.0.1/app/views/applicant/ApplicantUpsellCreateAccountView.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+public final class ApplicantUpsellCreateAccountView extends BaseHtmlView {
+
+  private final ApplicantLayout layout;
+
+  @Inject
+  public ApplicantUpsellCreateAccountView(ApplicantLayout layout) {
+    this.layout = checkNotNull(layout);
+  }
+
+  /** Renders a confirmation page for the applicant's submission. */
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/app/views/applicant/ApplicantUpsellCreateAccountView.java:28]: Suggestion (non-blocking): s/confirmation/signup/?

File: universal-application-tool-0.0.1/app/views/applicant/ApplicantUpsellCreateAccountView.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  public Content render(Http.Request request, String redirectTo, Messages messages) {
+    String title = "Account Creation Interstitial";
+    HtmlBundle bundle = layout.getBundle().setTitle(title);
+
+    ContainerTag content = div().withClasses(Styles.MX_16);
+    content.with(h2(messages.at("content.pleaseCreateAccount")));
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/app/views/applicant/ApplicantUpsellCreateAccountView.java:34]: Suggestion (non-blocking): mind moving these keys into MessageKey.java class?

**Review by hmhsu — APPROVED** [review]
Wow, there are so many redirects! *Inception*.

**Comment by ndmckinley** [comment]
Whew, thanks, this PR is a mess.  Maybe I'll come back to it on Monday!

File: universal-application-tool-0.0.1/app/controllers/applicant/RedirectController.java (hunk 2 above)
Comment (ndmckinley) [reply universal-application-tool-0.0.1/app/controllers/applicant/RedirectController.java:84]: That was a note to self, even, so removed.

File: universal-application-tool-0.0.1/app/controllers/applicant/RedirectController.java (hunk 3 above)
Comment (ndmckinley) [reply universal-application-tool-0.0.1/app/controllers/applicant/RedirectController.java:63]: Consider Register, yeah, that's better.

File: universal-application-tool-0.0.1/app/views/applicant/ApplicantUpsellCreateAccountView.java (hunk 4 above)
Comment (ndmckinley) [reply universal-application-tool-0.0.1/app/views/applicant/ApplicantUpsellCreateAccountView.java:29]: Done.

File: universal-application-tool-0.0.1/app/views/applicant/ApplicantUpsellCreateAccountView.java (hunk 5 above)
Comment (ndmckinley) [reply universal-application-tool-0.0.1/app/views/applicant/ApplicantUpsellCreateAccountView.java]: Done!

**Pull request merged**